### PR TITLE
[boschshc] Code enhancements and Sonar issue fixes

### PR DESCRIPTION
**Pull request by david-pace**
* replace call to deprecated HSBType::getRGB() with ColorUtil::hsbTosRgb()
* make constructors of abstract classes protected
* use instanceof with pattern matching (JEP 305)
* enhance switches with combined cases using comma-separated case expressions
* remove unnecessary public modifiers in unit tests

### CODE REVIEW

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/AbstractPowerSwitchHandler.java [hunk 1]
```diff
@@ -74,8 +74,8 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
         switch (channelUID.getId()) {
             case CHANNEL_POWER_SWITCH:
-                if (command instanceof OnOffType) {
-                    updatePowerSwitchState((OnOffType) command);
+                if (command instanceof OnOffType onOffType) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/AbstractPowerSwitchHandler.java:77]: Minor, perhaps this naming would be slightly easier to understand?
```suggestion
                if (command instanceof OnOffType onOffCommand) {
```

If you agree, consider all others as well.

**Review by jlaur — APPROVED** [review]
Thanks for the refactoring! LGTM - I have added one minor comment for consideration.

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/AbstractPowerSwitchHandler.java (hunk 1 above)
Comment (david-pace) [reply bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/AbstractPowerSwitchHandler.java]: Good idea 👍 I changed it consistently in all places (4 classes).

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smartbulb/SmartBulbHandler.java [hunk 2]
```diff
@@ -68,18 +69,18 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
-                if (command instanceof OnOffType) {
-                    updateBinarySwitchState((OnOffType) command);
+                if (command instanceof OnOffType onOffCommand) {
+                    updateBinarySwitchState(onOffCommand);
                 }
                 break;
             case CHANNEL_BRIGHTNESS:
-                if (command instanceof PercentType) {
-                    updateMultiLevelSwitchState((PercentType) command);
+                if (command instanceof PercentType percentType) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smartbulb/SmartBulbHandler.java:77]: For the same reason and consistency, I would do the same here:
```suggestion
                if (command instanceof PercentType percentCommand) {
```

File: bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smartbulb/SmartBulbHandler.java [hunk 3]
```diff
@@ -68,18 +69,18 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+                    updateMultiLevelSwitchState(percentType);
                 }
                 break;
             case CHANNEL_COLOR:
-                if (command instanceof HSBType) {
-                    updateColorState((HSBType) command);
+                if (command instanceof HSBType hsbType) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/smartbulb/SmartBulbHandler.java:82]: And here:
```suggestion
                if (command instanceof HSBType hsbCommand) {
```

**Review by jlaur — CHANGES_REQUESTED** [review]

**Review by GerdZanker — APPROVED** [review]
Thanks a lot for cleaning up the code

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**